monitor.py

```diff
@@ -156,103 +228,159 @@ def all(cls, _db, **kwargs):
             return 0
 
-    def identifier_query(self):
-        return self._db.query(Identifier)
+    def item_query(self):
+        """Find the items that need to be processed in the sweep.
+
+        :return: A query object.
+        """
+        # Start by getting everything in the table. 
+        qu = self._db.query(self.model_class)
+        if self.collection:
+            # Restrict to only those items associated with self.collection
+            # somehow.
+            qu = self.scope_to_collection(qu, self.collection)
+        qu = qu.order_by(self.model_class.id)
+        return qu
 
-    def process_batch(self, identifiers):
-        for i in identifiers:
-            self.process_identifier(i)
+    def scope_to_collection(self, qu, collection):
+        """Restrict the given query so that it only finds items
+        associated with the given collection.
 
-    def process_identifier(self, identifier):
+        :param qu: A query object.
+        :param collection: A Collection object, presumed to not be None.
+        """
+        raise NotImplementedError()
+    
+    def process_item(self, item):
+        """Do the work that needs to be done for a given item."""
         raise NotImplementedError()
 
-class SubjectSweepMonitor(IdentifierSweepMonitor):
 
-    def __init__(self, _db, name, subject_type=None, filter_string=None,
-                 batch_size=500):
-        super(SubjectSweepMonitor, self).__init__(
-            _db, name, batch_size=batch_size
+class IdentifierSweepMonitor(SweepMonitor):
+    """A Monitor that does some work for every Identifier."""
+    MODEL_CLASS = Identifier    
+
+    def scope_to_collection(self, qu, collection):
+        """Only find Identifiers licensed through the given Collection."""
+        return qu.join(Identifier.licensed_through).filter(
+            LicensePool.collection==collection
         )
-        self.subject_type = subject_type
-        self.filter_string = filter_string
 
-    def run_once(self, offset):
-        q = self.subject_query().filter(
-            Subject.id > offset).order_by(
-            Subject.id).limit(self.batch_size)
-        subjects = q.all()
-        if subjects:
-            self.process_batch(subjects)
-            return subjects[-1].id
-        else:
-            return 0
 
-    def subject_query(self):
+class SubjectSweepMonitor(SweepMonitor):
+    """A Monitor that does some work for every Subject."""
+    MODEL_CLASS = Subject
+
+    # It's usually easy to process a Subject, so make the batch size
+    # large.
+    DEFAULT_BATCH_SIZE = 500
+    
+    def __init__(self, _db, collection=None, subject_type=None,
+                 filter_string=None):
+        """Constructor.
+
+        :param collection: This value is ignored. There's no way
+            to run a SubjectSweepMonitor on a specific collection,
+            because there's no obvious connection between a Subject
+            and a Collection.
```

Comment:
I was thinking it would be better if all the Monitors shared a baseline set of constructor arguments so that they could take the same set of command-line arguments. Looking at the code we have now, I don't  think that's a strong argument. My other reason is that I can't think of a connection between Subject and Collection _now_, but I don't want to foreclose on the possibility,